[interfaces] use fixed date string in PythonSwig.cpp

olafhering opened the pull request:
The buildtime of an application has no meaning.
What might be interesting, and relevant, would be the source state, like a
git hash or versioned release.
Inserting such thing breaks what Debian sells as reproducible builds.

Signed-off-by: Olaf Hering <olaf@aepfle.de>

## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Rechi commented on xbmc/interfaces/python/PythonSwig.cpp.template:
```diff
@@ -916,7 +916,6 @@ namespace PythonBindings
 
    // constants
    PyModule_AddStringConstant(module, "__author__", "Team Kodi <http://kodi.tv>");
-   PyModule_AddStringConstant(module, "__date__", "${new Date().toString()}");
```
I would use `CCompileInfo::GetBuildDate`, which can be already set to a specific date via the build system, instead of removing it.

olafhering replied:
Done.

olafhering commented:
pull request updated

Rechi commented on xbmc/interfaces/python/PythonSwig.cpp.template:
```diff
@@ -749,6 +749,7 @@ Helper.getInsertNodes(module, 'begin').each { %>${Helper.unescape(it)}<% }
 
 #include <Python.h>
 #include <string>
+#include <CompileInfo.h>
```
wrong induce style (use `"`) as this is a Kodi header

Rechi reviewed (commented):
Commit message and PR title are wrong, because the date attribute isn't removed.

olafhering committed:
[interfaces] use fixed date string in PythonSwig.cpp

The buildtime of an application has no meaning.
What might be interesting, and relevant, would be the source state, like a
git hash or versioned release.
Inserting such thing breaks what Debian sells as reproducible builds.

Adjust the code to use the fixed string from CCompileInfo::GetBuildDate.

Signed-off-by: Olaf Hering <olaf@aepfle.de>

olafhering commented:
Thanks for the review.

Rechi commented:
wrong revision was build
jenkins build this please
